File: src/test/java/com/uwetrottmann/trakt5/services/EpisodesTest.java
/*
 * Copyright 2024 Uwe Trottmann
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.uwetrottmann.trakt5.services;

import com.uwetrottmann.trakt5.BaseTestCase;
import com.uwetrottmann.trakt5.TestData;
import com.uwetrottmann.trakt5.entities.Episode;
import com.uwetrottmann.trakt5.entities.Ratings;
import com.uwetrottmann.trakt5.entities.Stats;
import com.uwetrottmann.trakt5.enums.Extended;
import org.junit.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

public class EpisodesTest extends BaseTestCase {

    @Test
    public void test_summary() throws IOException {
        Episode episode = executeCall(getTrakt().episodes().summary(String.valueOf(TestData.SHOW_TRAKT_ID),
                TestData.EPISODE_SEASON,
                TestData.EPISODE_NUMBER, Extended.FULL));
        assertThat(episode).isNotNull();
        assertThat(episode.title).isEqualTo(TestData.EPISODE_TITLE);
        assertThat(episode.season).isEqualTo(TestData.EPISODE_SEASON);
        assertThat(episode.number).isEqualTo(TestData.EPISODE_NUMBER);
        assertThat(episode.ids.imdb).isEqualTo(TestData.EPISODE_IMDB_ID);
        assertThat(episode.ids.tmdb).isEqualTo(TestData.EPISODE_TMDB_ID);
        assertThat(episode.ids.tvdb).isEqualTo(TestData.EPISODE_TVDB_ID);
        assertThat(episode.runtime).isGreaterThan(0);
        assertThat(episode.comment_count).isGreaterThanOrEqualTo(0);
    }

    @Test
    public void test_comments() throws IOException {
        executeCall(getTrakt().episodes().comments(TestData.SHOW_SLUG, TestData.EPISODE_SEASON,
                TestData.EPISODE_NUMBER, 1, DEFAULT_PAGE_SIZE, null));
    }

    @Test
    public void test_ratings() throws IOException {
        Ratings ratings = executeCall(getTrakt().episodes().ratings(TestData.SHOW_SLUG, TestData.EPISODE_SEASON,
                TestData.EPISODE_NUMBER));
        assertRatings(ratings);
    }

    @Test
    public void test_stats() throws IOException {
        Stats stats = executeCall(
                getTrakt().episodes().stats(TestData.SHOW_SLUG, TestData.EPISODE_SEASON, TestData.EPISODE_NUMBER));
        assertStats(stats);
    }

}

File: src/test/kotlin/com/redissi/trakt/services/EpisodesTest.kt
package com.redissi.trakt.services

import com.redissi.trakt.BaseTestCase
import com.redissi.trakt.TestData
import com.redissi.trakt.TestRatings
import com.redissi.trakt.TestStats
import com.redissi.trakt.enums.Extended
import kotlinx.coroutines.runBlocking
import org.amshove.kluent.*
import org.junit.Test
import java.io.IOException

class EpisodesTest : BaseTestCase(), TestRatings, TestStats {
    @Test
    @Throws(IOException::class)
    fun `get summary for an episode`() = runBlocking<Unit> {
        val episode = trakt.episodes().summary(
            TestData.SHOW_TRAKT_ID.toString(),
            TestData.EPISODE_SEASON,
            TestData.EPISODE_NUMBER,
            Extended.FULL
        )

        episode.shouldNotBeNull()
        episode.title.shouldNotBeNull().shouldBeEqualTo(TestData.EPISODE_TITLE)
        episode.season.shouldNotBeNull().shouldBeEqualTo(TestData.EPISODE_SEASON)
        episode.number.shouldNotBeNull().shouldBeEqualTo(TestData.EPISODE_NUMBER)
        episode.ids.shouldNotBeNull()
        episode.ids!!.imdb.shouldBeEqualTo(TestData.EPISODE_IMDB_ID)
        episode.ids!!.tmdb.shouldBeEqualTo(TestData.EPISODE_TMDB_ID)
        episode.ids!!.tvdb.shouldBeEqualTo(TestData.EPISODE_TVDB_ID)
        episode.runtime.shouldNotBeNull().shouldBeGreaterThan(0)
        episode.commentCount.shouldNotBeNull().shouldBeGreaterOrEqualTo(0)
    }

    @Test
    @Throws(IOException::class)
    fun `get comments for an episode`() = runBlocking<Unit> {
        val comments = trakt.episodes().comments(
            TestData.SHOW_SLUG,
            TestData.EPISODE_SEASON,
            TestData.EPISODE_NUMBER,
            1,
            DEFAULT_PAGE_SIZE,
            null
        )
        comments.shouldNotBeNull()
        comments.size.shouldBeLessOrEqualTo(DEFAULT_PAGE_SIZE)
    }

    @Test
    @Throws(IOException::class)
    fun `get ratings for an episode`() = runBlocking {
        val ratings = trakt.episodes().ratings(TestData.SHOW_SLUG, TestData.EPISODE_SEASON, TestData.EPISODE_NUMBER)
        ratings.shouldNotBeNull()
        assertRatings(ratings)
    }

    @Test
    @Throws(IOException::class)
    fun `get stats for an episode`() = runBlocking {
        val stats = trakt.episodes().stats(TestData.SHOW_SLUG, TestData.EPISODE_SEASON, TestData.EPISODE_NUMBER)
        stats.shouldNotBeNull()
        assertStats(stats)
    }
}